lld/lld-c/LLDAsLibraryC.cpp

```diff
@@ -570,16 +581,21 @@ LLVMBool LLVMLinkEVM(LLVMMemoryBufferRef inBuffers[],
   assert(numInBuffers > 1);
   SmallVector<MemoryBufferRef> localInMemBufRefs(3);
   SmallVector<std::unique_ptr<MemoryBuffer>> localInMemBufs(3);
+
+  // TODO: verify that the object files contain sections with original
```

Comment:
If I understand correctly, we emit `.linker_symbol_name.` sections and place original name just to verify the names here? If that is correct, I would suggest not to emit these sections until we address this.